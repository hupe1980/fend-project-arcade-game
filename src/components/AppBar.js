import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import { Label, Image } from './common';
import withGame from './withGame';
import { IMAGES } from '../game';

const Container = styled.header`
  display: flex;
  flex-direction: row;
  width: 100%;
  justify-content: space-between;
  background-color: blue;
  color: white;
  padding-left: 10px;
  padding-right: 10px;
  margin-bottom: -20px;
`;

const ItemContainer = styled.div`
  display: flex;
  align-items: center;
`;

export function Level({ level }) {
  return (
    <ItemContainer>
      <Label>{`Level: ${level}`}</Label>
    </ItemContainer>
  );
}

Level.propTypes = {
  level: PropTypes.number.isRequired,
};

export function Player({ avatar, onChange }) {
  return (
    <ItemContainer>
      <Label>Player: </Label>
      <Image src={`./${avatar}`} alt="avatar" />
      <select onChange={onChange}>
        <option value={IMAGES['char-boy']}>Boy</option>
        <option value={IMAGES['char-cat-girl']}>Cat Girl</option>
        <option value={IMAGES['char-horn-girl']}>Horn Girl</option>
        <option value={IMAGES['char-pink-girl']}>Pink Girl</option>
        <option value={IMAGES['char-princess-girl']}>Princess Girl</option>
      </select>
    </ItemContainer>
  );
}

Player.propTypes = {
  avatar: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
};

export function Points({ points }) {
  return (
    <ItemContainer>
      <Label>{`${points} Points`}</Label>
    </ItemContainer>
  );
}

Points.propTypes = {
  points: PropTypes.number.isRequired,
};

export function Lives({ lives }) {
  return (
    <ItemContainer>
      <Image src={`./${IMAGES.heart}`} alt="heart" />
      <Label>{`x ${lives}`}</Label>
    </ItemContainer>
  );
}

Lives.propTypes = {
  lives: PropTypes.number.isRequired,
};

export function AppBar({ avatar, level, points, lives, changeAvatar }) {
  return (
    <Container>
      <Level level={level} />
      <Player avatar={avatar} onChange={e => changeAvatar(e.target.value)} />
      <Points points={points} />
      <Lives lives={lives} />
    </Container>
  );
}

AppBar.propTypes = {
  avatar: PropTypes.string.isRequired,
  level: PropTypes.number.isRequired,
  points: PropTypes.number.isRequired,
  lives: PropTypes.number.isRequired,
  changeAvatar: PropTypes.func.isRequired,
};

export default withGame(AppBar);
