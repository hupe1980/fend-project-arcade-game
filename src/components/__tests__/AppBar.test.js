import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { AppBar, Level, Player, Points, Lives } from '../AppBar';

describe('AppBar', () => {
  it('renders without crashing (AppBar)', () => {
    shallow(
      <AppBar
        level={999}
        avatar={'image_url'}
        points={999}
        lives={999}
        changeAvatar={i => i}
      />,
    );
  });

  describe('Level', () => {
    it('renders correctly', () => {
      const wrapper = shallow(<Level level={999} />);
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });

  describe('Player', () => {
    it('renders correctly', () => {
      const wrapper = shallow(
        <Player avatar={'image_url'} onChange={i => i} />,
      );
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });

  describe('Points', () => {
    it('renders correctly', () => {
      const wrapper = shallow(<Points points={999} />);
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });

  describe('Lives', () => {
    it('renders correctly', () => {
      const wrapper = shallow(<Lives lives={999} />);
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});
