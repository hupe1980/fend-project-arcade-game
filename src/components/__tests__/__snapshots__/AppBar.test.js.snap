// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppBar Level renders correctly 1`] = `
<styled.div>
  <styled.span>
    Level: 999
  </styled.span>
</styled.div>
`;

exports[`AppBar Lives renders correctly 1`] = `
<styled.div>
  <styled.img
    alt="heart"
    src="./images/heart.png"
  />
  <styled.span>
    x 999
  </styled.span>
</styled.div>
`;

exports[`AppBar Player renders correctly 1`] = `
<styled.div>
  <styled.span>
    Player: 
  </styled.span>
  <styled.img
    alt="avatar"
    src="./image_url"
  />
  <select
    onChange={[Function]}
  >
    <option
      value="images/char-boy.png"
    >
      Boy
    </option>
    <option
      value="images/char-cat-girl.png"
    >
      Cat Girl
    </option>
    <option
      value="images/char-horn-girl.png"
    >
      Horn Girl
    </option>
    <option
      value="images/char-pink-girl.png"
    >
      Pink Girl
    </option>
    <option
      value="images/char-princess-girl.png"
    >
      Princess Girl
    </option>
  </select>
</styled.div>
`;

exports[`AppBar Points renders correctly 1`] = `
<styled.div>
  <styled.span>
    999 Points
  </styled.span>
</styled.div>
`;
